internal/handlers/todo.go

```diff
@@ -0,0 +1,61 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/shreyas1307/go-server/internal/store"
+	"github.com/shreyas1307/go-server/internal/store/models"
+)
+
+type TodoHandler struct {
```

Comment:
So now let's imagine a scenario, interfaces can be nil. And in go there's no protection from this:

```go
h := TodoHandler{}
```

What you really want is:
```go
h := TodoHandler{myStore}
```

One way around this (and only when it really matters) is "constructors". I put it in quotes because they're not real constructors, but colloquially we call them constructors and you'll often see it in packages. So I'd add something like:

```go
func NewTodoHandler(store store.ServerStore) (*TodoHandler, error) {
      if store == nil {
              return nil, errors.New("store cannot be nil")
      }

      return &TodoHandler{store}
}
```